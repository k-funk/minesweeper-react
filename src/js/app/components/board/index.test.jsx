import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import Board, { generateUniqueRandomInts, plantMine } from './index';


describe('outputs the expected tree when', () => {
  let wrapper;

  test('5x5 board with 3 mines', () => {
    // FIXME: need to mock `initBoardData` so that the snapshots match on each run
    wrapper = shallow((
      <Board width={5} height={5} mines={3} />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('generateUniqueRandomInts', () => {
  const fn = generateUniqueRandomInts;
  const lowerLimit = 0;
  let ints;
  let amount;
  let upperLimit;

  test('valid values', () => {
    amount = 3;
    upperLimit = 10;
    ints = fn(amount, upperLimit);
    expect(ints).toHaveLength(amount);
  });

  test('a full set', () => {
    amount = 10;
    upperLimit = 10;
    ints = fn(amount, upperLimit);
    expect(ints).toHaveLength(amount);
  });

  test('beyond the upper limit does nothing extra', () => {
    amount = 11;
    upperLimit = 10;
    ints = fn(amount, upperLimit);
    expect(ints).toHaveLength(10);
  });

  afterEach(() => {
    expect(Math.max(...ints)).toBeLessThan(upperLimit);
    expect(Math.min(...ints)).toBeGreaterThanOrEqual(lowerLimit);
  });
});

test('plantMine', () => {
  const board = [
    [{ isMine: false }, { isMine: false }],
    [{ isMine: false }, { isMine: false }],
  ];
  const index = 2;

  plantMine(board, index);

  expect(board).toEqual([
    [{ isMine: false }, { isMine: false }],
    [{ isMine: true }, { isMine: false }],
  ]);
});
