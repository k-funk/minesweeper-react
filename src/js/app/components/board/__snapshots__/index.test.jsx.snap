// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when 5x5 board with 3 mines 1`] = `
<div
  className=""
>
  <styled.div
    key="row-0"
  >
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-0,0"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-0,1"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": true,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-0,2"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-0,3"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-0,4"
      onClick={[Function]}
    />
  </styled.div>
  <styled.div
    key="row-1"
  >
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-1,0"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-1,1"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-1,2"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-1,3"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-1,4"
      onClick={[Function]}
    />
  </styled.div>
  <styled.div
    key="row-2"
  >
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-2,0"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": true,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-2,1"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-2,2"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-2,3"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-2,4"
      onClick={[Function]}
    />
  </styled.div>
  <styled.div
    key="row-3"
  >
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-3,0"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-3,1"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-3,2"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-3,3"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-3,4"
      onClick={[Function]}
    />
  </styled.div>
  <styled.div
    key="row-4"
  >
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-4,0"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": true,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-4,1"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-4,2"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-4,3"
      onClick={[Function]}
    />
    <Cell
      cellData={
        Object {
          "isFlagged": false,
          "isMine": false,
          "isRevealed": false,
          "neighbors": 0,
        }
      }
      key="cell-4,4"
      onClick={[Function]}
    />
  </styled.div>
</div>
`;
